pkg/ipamd/ipamd.go

```diff
@@ -300,9 +311,20 @@ func New(k8sapiClient kubernetes.Interface, eniConfig *eniconfig.ENIConfigContro
 	c.warmENITarget = getWarmENITarget()
 	c.warmIPTarget = getWarmIPTarget()
 	c.minimumIPTarget = getMinimumIPTarget()
+	c.warmPrefixTarget = getWarmPrefixTarget()
 
 	c.disableENIProvisioning = disablingENIProvisioning()
 	c.enablePodENI = enablePodENI()
+
+	hypervisorType, err := c.awsClient.GetInstanceHypervisorFamily()
+	if err != nil {
+		log.Error("Failed to get hypervisor type")
+		return nil, err
+	}
+	if hypervisorType != "nitro" && c.enableIpv4PrefixDelegation {
+		log.Error("Hypervisor is not nitro based and prefix delegation is not supported")
```

Comment:
Hypervisor is not nitro based and prefix delegation is not supported -> Prefix delegation is not supported non-nitro instances %s ..